### PR TITLE
Client-side Bitcoin Transactions

### PR DESCRIPTION
**Pull request by kantai**
This performs bitcoin transactions client-side for the route

```
/wallet/send
```

To test this in regtest, I recommend firing up a regtest container:

```
$ IMAGE=$(docker run -dt -p 6270:6270 -p 16269:16269 -p 18332:18332 -e BLOCKSTACK_TEST_CLIENT_RPC_PORT
=6270 -e BLOCKSTACK_TEST_CLIENT_BIND=0.0.0.0 -e BLOCKSTACK_TEST_BITCOIND_ALLOWIP=172.17.0.0/16 quay.io/blockstack/inte
grationtests:master blockstack-test-scenario --interactive 2 blockstack_integration_tests.scenarios.browser_env)
$ docker logs -f $IMAGE | grep -q 'Test finished'
```

Then run the `dev-proxy` and `dev` npm targets:

```
$ npm i && npm run dev-proxy & npm run dev
```

Then open `localhost:3000/#coreAPIPassword=blockstack_integration_test_api_password&regtest=1`

After initializing an account, fund your regtest wallet from the `/send-core` endpoint.

Then you can send things around from the `/send/` endpoint.

To test in mainnet, just send a transaction from the `/send/` endpoint.

## Things that are different

If you enter a BTC amount _greater_ than the amount of bitcoin the wallet has, it will send the _max_ amount, not error. Do we want the old behavior instead, where it would error out? (For what it's worth, I found that old behavior really annoying, because I could never actually zero out my browser wallet).

## Things someone should look at

I don't _really_ know the right way to dispatch actions in reactJS, I just kind of fake knowledge of that. This method dispatches some `coreWalletX` actions. I don't think that should be a problem, but maybe it is, so someone should look at that.

Issue #1239

### CODE REVIEW

File: app/js/account/store/account/actions.js [hunk 1]
```diff
@@ -255,6 +257,41 @@ function resetCoreWithdrawal() {
...
+        const myNet = config.network
+        logger.trace(`Broadcast btc spend with tx hex: ${transactionHex}`)
+        return myNet.broadcastTransaction(transactionHex)
+      })
+      .then(() => {
+        dispatch(withdrawCoreBalanceSuccess())
+      })
+      .catch((error) => {
+        logger.error('withdrawBitcoinClientSide: error generating and broadcasting', error)
+        dispatch(withdrawCoreBalanceError(error))
```
[the remark this answers is not in the document]
Comment (kantai) [reply app/js/account/store/account/actions.js]: Okay, noted. Will change that.

**Comment by kantai** [comment]
Thanks for the review @cwackerfuss! The linked loom video appears to be a different one than you expected?

**Comment by cwackerfuss** [comment]
Updated the link @kantai -- thanks for the heads up!

**Comment by cwackerfuss** [comment]
The first problem I ran into in the video was a cryptic message when I try to pass a wallet id thats not actually a valid ID. Then once the transaction goes through, the UI doesn't reflect that a transaction / change on my wallet is pending. It'd be nice to temporarily hide the wallet value until the new number appears.

**Comment by kantai** [comment]
Okay cool -- I think @cwackerfuss is going to work on those UX improvements in a separate PR.

So should I merge to `develop` @cwackerfuss, or do you think another person should test some transactions?

**Comment by cwackerfuss** [comment]
Not necessary if you're comfortable merging. I'm sure you've tested it several times as well.

**Comment by kantai** [comment]
Cool, thanks again @cwackerfuss!

Merging...

**Pull request merged**